src/main/java/ua/lviv/iot/novaPoshtaAPI/controller/CourierController.java

```diff
@@ -0,0 +1,40 @@
+package ua.lviv.iot.novaPoshtaAPI.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import ua.lviv.iot.novaPoshtaAPI.model.Courier;
+import ua.lviv.iot.novaPoshtaAPI.service.CourierService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/couriers")
+public class CourierController {
+
+    @Autowired
+    CourierService courierService;
+
+    @GetMapping
+    public List<Courier> getAllCouriers() {
+        return courierService.getAllCouriers();
+    }
+
+    @GetMapping("/{courierId}")
+    public Courier getCourierById(@PathVariable Long courierId) {
+        return courierService.getCourierById(courierId);
+    }
+
+    @RequestMapping("/{courierId}/deliver/{parcelId}")
```

Comment:
this method is operating with parcel, not courier. So it should be in ParcelController (together with methods to create the parcel as well)

Reply:
See the reply above. The logic here has been changed a bit from a traditional one